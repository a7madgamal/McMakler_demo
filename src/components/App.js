import React, {Component} from 'react';
import logo from '../style/logo.svg';
import List from './List';
import McMaklerApi from '../models/McMaklerApi';

import '../style/App.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      items: [],
      status: 'loading'
    };
  }
  componentDidMount() {
    let mcMaklerApi = new McMaklerApi();
    mcMaklerApi
      .fetchFakeAds(10)
      .then(adsList => {
        this.setState({
          items: adsList,
          status: 'ok'
        });
      })
      .catch(ex => {
        this.setState({
          status: 'error',
          items: []
        });
      });
  }
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to React</h2>
        </div>
        {this.state.status === 'ok'
          ? <List items={this.state.items} />
          : <div className="error-message">Oops, something went wrong!</div>}
      </div>
    );
  }
}

export default App;
