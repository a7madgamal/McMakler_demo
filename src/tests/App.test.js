import React from 'react';
import renderer from 'react-test-renderer';
import App from '../components/App'
it('renders correctly: no props', () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
});
