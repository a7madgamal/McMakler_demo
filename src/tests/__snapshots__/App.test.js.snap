// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly: no props 1`] = `
<div
  className="App"
>
  <div
    className="error-message"
  >
    Oops, something went wrong!
  </div>
</div>
`;
